v8/third_party/heap/mmtk/mmtk.cc

```diff
@@ -176,17 +180,23 @@ bool Heap::IsValidHeapObject(HeapObject object) {
   return is_live_object(reinterpret_cast<void*>(object.address()));
 }
 
+// Access to the heap object iterator is assumed to be single-threaded.
```

Comment:
Is this still true if we have multiple isolates?

Reply:
No. For multiple isolates, the heap iterator needs a rewrite. For now, my goal is to just make single isolate programs/tests work.